modules/data.remote/R/call_remote_process.R

```diff
@@ -0,0 +1,400 @@
+
+  reticulate::import_from_path("remote_process", file.path("..", "inst"))
+  reticulate::source_python(file.path("..", "inst", "remote_process.py"))
+  
+  input_file <- NULL
+  stage_get_data <- NULL
+  stage_process_data <- NULL
+  raw_merge <- NULL
+  pro_merge <- NULL
+  existing_raw_file_path <- NULL
+  existing_pro_file_path <- NULL
+  
+  # extract the variables from the settings list
+  siteid <- settings$remotedata$siteid
+  sitename <- settings$remotedata$sitename
+  raw_mimetype <- settings$remotedata$raw_mimetype
+  raw_formatname <- settings$remotedata$raw_formatname
+  geofile <- settings$remotedata$geofile
+  outdir <- settings$remotedata$outdir
+  start <- settings$remotedata$start
+  end <- settings$remotedata$end
+  source <- settings$remotedata$source
+  collection <- settings$remotedata$collection
+  scale <- settings$remotedata$scale
+  if(!is.null(scale)){
+  scale <- as.double(settings$remotedata$scale)
+  scale <- format(scale, nsmall = 1)
+  }
+  projection <- settings$remotedata$projection
+  qc <- settings$remotedata$qc
+  if(!is.null(qc)){
+  qc <- as.double(settings$remotedata$qc)
+  qc <- format(qc, nsmall = 1)
+  }
+  algorithm <- settings$remotedata$algorithm
+  credfile <- settings$remotedata$credfile
+  pro_mimetype <- settings$remotedata$pro_mimetype
+  pro_formatname <- settings$remotedata$pro_formatname
+  out_get_data <- settings$remotedata$out_get_data
+  out_process_data <- settings$remotedata$out_process_data
+  
+  
+  dbcon <- PEcAn.DB::db.open(settings$database$bety)
+  flag <- 0
+
+  # collection dataframe used to map Google Earth Engine collection names to their PEcAn specific names
+  collection_lut <- data.frame(stringsAsFactors=FALSE,
+                          original_name = c("LANDSAT/LC08/C01/T1_SR", "COPERNICUS/S2_SR", "NASA_USDA/HSL/SMAP_soil_moisture"),
+                          pecan_code = c("l8", "s2", "smap")
+  )
+  getpecancode <- collection_lut$pecan_code
+  names(getpecancode) <- collection_lut$original_name
+  
+  if(source == "gee"){
+  collection = unname(getpecancode[collection])
+  }
+
+  req_start <- start
+  req_end <- end
+  
+  raw_file_name = construct_raw_filename(sitename, source, collection, scale, projection, qc)
+  
+  # check if any data is already present in the inputs table
+  existing_data <- PEcAn.DB::db.query(paste0("SELECT * FROM inputs WHERE site_id=", siteid), dbcon)
+  if(nrow(existing_data) >= 1){
+    
+    # if processed data is requested, example lai
+    if(!is.null(out_process_data)){
+      
+      # construct processed file name
+      pro_file_name = paste0(sitename, "_", out_process_data, "_", algorithm)
+      
+      # check if processed file exists
+      if(nrow(pro_check <- PEcAn.DB::db.query(sprintf("SELECT inputs.id, inputs.site_id, inputs.name, inputs.start_date, inputs.end_date, dbfiles.file_path FROM inputs INNER JOIN dbfiles ON inputs.name=dbfiles.file_name AND inputs.name LIKE '%s%%';", pro_file_name), dbcon)) == 1){
```

Comment:
feel like this line is reinventing `dbfile.input.check`

Reply:
I don't think so because here as you can see, I'm searching using the filename which contains more info about the scale, projection and qc. Using `dbfile.input.check` wouldn't fulfil my motive unless a new format has to be created whenever scale, qc, etc are changed.